src/test/unit_tests/test_notifications.js

```diff
@@ -0,0 +1,203 @@
+mocha.describe('notifications', function() {
+
+    this.timeout(40000); // eslint-disable-line no-invalid-this
+    let s3;
+
+    describe('notifications', () => {
+
+        mocha.before(async function() {
+
+            const admin_keys = (await rpc_client.account.read_account({ email: EMAIL, })).access_keys;
+            s3 = generate_s3_client(admin_keys[0].access_key.unwrap(),
+                admin_keys[0].secret_key.unwrap(),
+                coretest.get_http_address());
+
+            //create http connect file
+            fs.mkdirSync(tmp_connect, {recursive: true});
+            fs.writeFileSync(http_connect_path, JSON.stringify(http_connect));
+
+            fs.mkdirSync(notif_logs_path, {recursive: true});
+
+            await s3.createBucket({
+                Bucket,
+            });
+
+            http_server = http.createServer(async function(req, res) {
+                const chunks = [];
+
+                for await (const chunk of req) {
+                    chunks.push(chunk);
+                }
+
+                const input = Buffer.concat(chunks);
+                const notif = JSON.parse(input.toString());
+
+                if (notif !== "test notification") {
+                    assert.strictEqual(notif.Records[0].s3.bucket.name, expected_bucket, 'wrong bucket name in notification');
+                    assert.strictEqual(notif.Records[0].eventName, expected_event_name, 'wrong event name in notification');
+                }
+
+                res.writeHead(200, {'Content-Type': 'text/plain'});
+                res.end();
+                server_done = true;
+            }).listen(9999);
+        });
+
+        mocha.after(() => {
+            http_server.close();
+        });
+
+        mocha.it('set/get notif conf s3ops', async () => {
+            await s3.putBucketNotificationConfiguration({
+                Bucket,
+                NotificationConfiguration: {
+                    TopicConfigurations: [{
+                        "Id": "system_test_http_no_event",
+                        "TopicArn": http_connect_filename,
+                    }],
+                },
+            });
+
+            const get = await s3.getBucketNotificationConfiguration({Bucket});
+            assert.strictEqual(get.TopicConfigurations[0].Id, 'system_test_http_no_event');
+        });
+
+        mocha.it('simple notif put', async () => {
+            await s3.putObject({
+                Bucket,
+                Key: 'f1',
+                Body: 'this is the body',
+            });
+
+            await notify_await_result({bucket: Bucket, event_name: 'ObjectCreated:Put'});
+        });
+
+
+        mocha.it('simple notif delete', async () => {
+            await s3.deleteObject({
+                Bucket,
+                Key: 'f1',
+            });
+
+            await notify_await_result({bucket: Bucket, event_name: 'ObjectRemoved:Delete'});
+        });
+
+
+        mocha.it('event notif', async () => {
```

Comment:
I didn't understand the title here (we already tested put and delete notification), maybe you add more explanation here?